src/TwigComponent/src/ComponentRenderer.php

```diff
@@ -148,6 +148,14 @@ private function exposedVariables(object $component, bool $exposePublicProps): \
                 throw new \LogicException(sprintf('Cannot use %s on methods with required parameters (%s::%s).', ExposeInTemplate::class, $component::class, $method->name));
             }
 
+            if ($attribute->destruct) {
```

Comment:
We need similar code near the top of the function when looping over properties. You could have a property with the attribute on it and this new option.